core-primitives/substrate-sgx/environmental/src/lib.rs

```diff
@@ -125,14 +125,7 @@ pub fn with<T: ?Sized, R, F: FnOnce(&mut T) -> R>(
 		// We always use the `last` element when we want to access the
 		// currently set global.
 		let last = r.borrow().last().cloned();
-		match last {
-			Some(ptr) => unsafe {
-				// safe because it's only non-zero when it's being called from using, which
-				// is holding on to the underlying reference (and not using it itself) safely.
-				Some(mutator(&mut **ptr.borrow_mut()))
-			},
-			None => None,
-		}
+		last.map(|ptr| unsafe { mutator(&mut **ptr.borrow_mut()) })
```

Comment:
I'd like to keep the comment.

Reply:
Good point! Is it alright like that, or should I keep the old version and add a clippy exception?

Reply:
It is fine for me as it is. :)